tests/expectations/test_dataclass_serializable_dot_dict_pattern.py

```diff
@@ -0,0 +1,507 @@
+    "Objects of this type can be nested"
+
+    my_C = MyClassC(
+        alpha_var=20,
+        beta_var=MyEnum("x"),
+        A_list=[
+            MyClassA(
+                foo="A-1",
+                bar=101,
+            ),
+            MyClassA(
+                **{
+                    "foo": "A-2",
+                    "bar": 102,
+                }
+            ),
+        ],
+        B_list=[
+            MyClassB(
+                **{
+                    "foo": "B-1",
+                    "bar": 201,
+                    "baz": ["a", "b", "c"],
+                    "qux": -100,
+                    "quux": 43,
+                }
+            )
+        ],
+    )
+
+    # Demonstrate that we can access sub-objects using a mix of dict and dot notation:
+    assert my_C.A_list[0].foo == "A-1"
+    assert my_C["A_list"][1].bar == 102
+    assert my_C["B_list"][0]["quux"] == 43
+
+    # Note: we don't currently support dot notation access within lists: `assert my_C["A_list"].1.bar == 102`
+
+    # Demonstrate that we can access Enum sub-objects
+    assert my_C["beta_var"] == MyEnum("x")
+
+
+def test_to_dict_works_recursively():
+    "the .to_dict method works recursively on both DotDicts and Enums"
+
+    my_C = MyClassC(
+        alpha_var=20,
+        beta_var=MyEnum("x"),
+        A_list=[
+            MyClassA(
+                foo="A-1",
+                bar=101,
+            ),
+            MyClassA(
+                **{
+                    "foo": "A-2",
+                    "bar": 102,
+                }
+            ),
+        ],
+        B_list=[
+            MyClassB(
+                **{
+                    "foo": "B-1",
+                    "bar": 201,
+                    "baz": ["a", "b", "c"],
+                    "qux": -100,
+                    "quux": 43,
+                }
+            )
+        ],
+        enum_list=[
+            MyEnum("x"),
+            MyEnum("y"),
+            MyEnum("z"),
+        ]
+    )
+
+    C_dict = my_C.to_dict()
+
+    # Make sure it's a dictionary, not a DictDot
+    assert type(C_dict) == dict
+    assert isinstance(C_dict, DictDot) == False
+    # Dictionaries don't support dot notation.
+    with raises(AttributeError):
+        C_dict.A_list
+
+    assert type(C_dict["A_list"][0]) == dict
+    assert type(C_dict["B_list"][0]) == dict
+
+    assert C_dict == {
+        "alpha_var": 20,
+        "beta_var": "x",
+        "A_list": [
+            {
+                "foo": "A-1",
+                "bar": 101,
+            },
+            {
+                "foo": "A-2",
+                "bar": 102,
+            },
+        ],
+        "B_list": [
+            {
+                "foo": "B-1",
+                "bar": 201,
+                "baz": ["a", "b", "c"],
+                "qux": -100,
+                "quux": 43,
+            }
+        ],
+        "enum_list": [
+            "x",
+            "y",
+            "z",
+        ]
+    }
+
+
+def test_immutability():
```

Comment:
I feel this test is verifying the python language (and @dataclass decorator), not our code

Reply:
Removing...